incompatibility/apiset.go

```diff
@@ -44,13 +42,66 @@ func AggregateAnalysis(analysis ...*ApiSetIncompatibility) *ApiSetIncompatibilit
 	aggAnalysis := NewAnalysis()
 	for _, analysisObj := range analysis {
 		aggAnalysis.OpenApiFiles += analysisObj.OpenApiFiles
+		aggAnalysis.IncompatibleFiles += analysisObj.IncompatibleFiles
+		aggAnalysis.AnalysisPerIncompatibility =
+			mergeIncompatibilityAnalysis(aggAnalysis.AnalysisPerIncompatibility, analysisObj.AnalysisPerIncompatibility)
 	}
 	return aggAnalysis
 }
 
-// AggregateIncompatiblities aggrates information from individual incompatibilities into an
-// existing analysis object
-func AggregateIncompatibilities(analysis *ApiSetIncompatibility, incompatibilities ...*Incompatibility) *ApiSetIncompatibility {
+// FormAnalysis creates an analysis object from a single IncompatibilityReport
+func FormAnalysis(report *IncompatibilityReport, uniqueFilePath string) *ApiSetIncompatibility {
```

Comment:
A file path is inherently unique.  Can we remove that from the parameter name?

Reply:
now filePath